dvc/objects/diff.py

```diff
@@ -0,0 +1,135 @@
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, List, Optional, Tuple
+
+if TYPE_CHECKING:
+    from .file import HashFile
+
+ADD = "add"
+MODIFY = "modify"
+DELETE = "delete"
+UNCHANGED = "unchanged"
+
+
+@dataclass
+class KeyObjPair:
```

Comment:
`KeyObjPair` looks a bit weird, maybe `FileChange`/`DiffFile`, etc. as this relates to a single file.

Reply:
Yeah, didn't want to use `file` there, as it could be used for trees too in the near future (and actually now too when ROOT used to be a file but now a tree), but that's still better than `pair` and we can adjust it later anyway. Will adjust.